SampleUserControlLibrary/SubscriptionKeyPage.xaml.cs

```diff
@@ -149,36 +204,70 @@ private void SaveSubscriptionKeyToIsolatedStorage(string subscriptionKey)
         }
 
         /// <summary>
-        /// Handles the Click event of the subscription key save button.
+        /// Saves the subscription endpoint to isolated storage.
+        /// </summary>
+        /// <param name="subscriptionEndpoint">The subscription key.</param>
+        private void SaveSubscriptionEndpointToIsolatedStorage(string subscriptionEndpoint)
+        {
+            using (IsolatedStorageFile isoStore = IsolatedStorageFile.GetStore(IsolatedStorageScope.User | IsolatedStorageScope.Assembly, null, null))
+            {
+                using (var oStream = new IsolatedStorageFileStream(_isolatedStorageSubscriptionEndpointFileName, FileMode.Create, isoStore))
+                {
+                    using (var writer = new StreamWriter(oStream))
+                    {
+                        writer.WriteLine(subscriptionEndpoint);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Set a default endpoint when there is no legal endpoint value
+        /// </summary>
+        /// <param name="endpoint"></param>
+        public void SetSubscriptionEndpoint(string endpoint){
+            string subscriptionEndpoint = null;
+            subscriptionEndpoint = GetSubscriptionEndpointFromIsolatedStorage();
```

Comment:
In fact, I've tried, but we cannot simply set the public string SubscriptionEndpoint, if I change the setter function, it throws a null pointer exception(may be caused by binding). It's a big problem and I use my own way to handle this.